### PR TITLE
Add OIDC scopes for email & profile retrieval

### PR DESCRIPTION
**Pull request by elespike**
Initial discussion at #276

### CODE REVIEW

**Comment by elespike** [comment]
Force push was to add signoff.

**Comment by SuperSandro2000** [comment]
I think it complains about the author of the first commit.

**Comment by elespike** [comment]
Yep, my bad. Forgot to set git config.

**Comment by elespike** [comment]
Ah I see the author of the very first one... seems ok now, if you don't mind my pseudonym there. =p

File: lib/config/hackmdEnvironment.js [hunk 1]
```diff
@@ -81,7 +81,8 @@ module.exports = {
     tokenURL: process.env.HMD_OAUTH2_TOKEN_URL,
     authorizationURL: process.env.HMD_OAUTH2_AUTHORIZATION_URL,
     clientID: process.env.HMD_OAUTH2_CLIENT_ID,
-    clientSecret: process.env.HMD_OAUTH2_CLIENT_SECRET
+    clientSecret: process.env.HMD_OAUTH2_CLIENT_SECRET,
+    scope: process.env.HMD_OAUTH2_SCOPE
```
Comment (elespike) [inline lib/config/hackmdEnvironment.js]: Wasn't sure if this one was necessary.

**Comment by SuperSandro2000** [comment]
I personally don't mind but the codimd project does. See [contributing](https://github.com/codimd/server/blob/master/CONTRIBUTING.md#sign-your-work). Sorry about that.

**Comment by elespike** [comment]
Gotcha, I thought that was referring only to the sign-offs. Let me figure out how to amend that.

**Comment by elespike** [comment]
There. Ugh, all the force pushes. -_-
after all this time, still a git n00b =)

**Comment by SISheogorath** [comment]
Tip: If you need to sign-off a bunch off commits you can use `git rebase master --sign-off` this will rebase your commits to latest master and makes sure all have a proper sign-off line :)

**Comment by SISheogorath** [comment]
Thanks for your work! Moved it to 2.0 :) That will be a great release!

**Comment by oupala** [comment]
Do you have any idea when the 2.0 release should be available?

If the planned date is too far in the futur, do you think it could be possible de release a 1.6.1 version so that the OIDC connection could be fixed?

This bug is really preventing from using CodiMD with an OIDC authentication.

**Comment by SuperSandro2000** [comment]
> Do you have any idea when the 2.0 release should be available?

It will probably take a few more months. Maybe fall, maybe later or earlier. We are not to sure yet.

We will probably do a bugfix only 1.6.1 before that as we collected some things we want to fix before 2.0.

**Pull request merged**